include/lbann/data_readers/sample_list_impl.hpp

```diff
@@ -378,6 +383,92 @@ ::all_gather_archive(const std::string &archive,
   return;
 }
 
+template <typename sample_name_t>
+inline void sample_list<sample_name_t>
+::all_gather_archive_new(const std::string &archive,
+                     std::vector<std::string>& gathered_archive,
+                     lbann_comm& comm) {
+
+  // there's commented out code below to deal with the case where
+  // archive.size() > INT_MAX; but for now let's assume we won't
+  // encounter that (which is true for the 100M JAG set)
+  size_t n = archive.size();
+  if (n > INT_MAX) {
```

Comment:
I would recommend `#include <limits>` and then replace `INT_MAX` with `std::numeric_limits<int>::max()` (or define your own `constexpr int` that has value equal to `std::numeric_limits<int>::max()`. E.g.:
```cpp
int constexpr max_int = std::numeric_limits<int>::max();
```
`numeric_limits` is the C++ idiom for expressing this information, and since the result (of this particular function, anyway) is `constexpr`, it can be used anywhere the macro would be (templates, case statements, array dims, etc).

Reply:
In view of Tim's comment,
```
using int_type = /*whatever*/;
int_type constexpr max_value = std::numeric_limits<int_type>::max();
```
(Or template the function on the integer type and then just don't worry about it...)

Reply:
Also also, comparing `size_t` to `int` will give a compiler warning from worthwhile compilers.